### PR TITLE
Support the ES7

### PR DESCRIPTION
**Pull request by stokarenko**
Adds the support for both ES6 and ES7

### CODE REVIEW

File: lib/searchkick/index_options.rb [hunk 1]
```diff
@@ -4,21 +4,29 @@ def index_options
       options = @options
       language = options[:language]
       language = language.call if language.respond_to?(:call)
-      index_type = options[:_type]
```
Comment (stokarenko) [inline lib/searchkick/index_options.rb:7]: Whole the file has been copied from upstream, considering that we never changed it

**Review by edtaupier — APPROVED** [review]

**Review by stevehodges — APPROVED** [review]

File: lib/searchkick/results.rb [hunk 2]
```diff
@@ -132,7 +132,13 @@ def entry_name(options = {})
...
 
     def total_count
-      options[:total_entries] || response["hits"]["total"]
+      if options[:total_entries]
+        options[:total_entries]
+      elsif response["hits"]["total"].is_a?(Hash)
+        response["hits"]["total"]["value"]
+      else
+        response["hits"]["total"]
+      end
```
Comment (scifisamurai) [inline lib/searchkick/results.rb:141]: Same as upstream as of 4.6.3 ( last version to support Elasticsearch 6 )

File: lib/searchkick/record_data.rb [hunk 3]
```diff
@@ -40,14 +40,17 @@ def document_type(ignore_type = false)
     def record_data
       data = {
         _index: index.name,
-        _id: search_id,
-        _type: document_type
+        _id: search_id
       }
-      data[:_routing] = record.search_routing if record.respond_to?(:search_routing)
+      data[:_type] = document_type if Searchkick.server_below?("7.0.0")
```
Comment (scifisamurai) [inline lib/searchkick/record_data.rb:45]: Above is same as upstream as of 4.6.3 ( except upstream uses a slightly different version check )

File: lib/searchkick/query.rb [hunk 4]
```diff
@@ -168,7 +168,7 @@ def handle_response(response)
     end
 
     def retry_misspellings?(response)
-      @misspellings_below && response["hits"]["total"] < @misspellings_below
+      @misspellings_below && Searchkick::Results.new(searchkick_klass, response).total_count < @misspellings_below
```
Comment (scifisamurai) [inline lib/searchkick/query.rb:171]: Above is same as upstream as of 4.6.3

**Pull request merged**